modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -49,9 +50,20 @@ func (k Keeper) OnChanOpenInit(
 		return err
 	}
 
-	activeChannelID, found := k.GetOpenActiveChannel(ctx, connectionHops[0], portID)
+	activeChannelID, found := k.GetActiveChannelID(ctx, connectionHops[0], portID)
 	if found {
-		return sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s", activeChannelID, portID)
+		channel, found := k.channelKeeper.GetChannel(ctx, portID, activeChannelID)
+		if !found {
+			panic(fmt.Sprintf("active channel mapping set for %s but channel does not exist in channel store", activeChannelID))
+		}
+
+		if channel.State == channeltypes.OPEN {
+			return sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s", activeChannelID, portID)
+		}
+
+		if !icatypes.IsPreviousMetadataEqual(version, metadata) {
```

Comment:
I might be missing something, but the `version` that is passed here, isn't it the same that is unmarshalled to `metadata` above? So here:

```
var metadata icatypes.Metadata
if err := icatypes.ModuleCdc.UnmarshalJSON([]byte(version), &metadata); err != nil {
    return sdkerrors.Wrapf(icatypes.ErrUnknownDataType, "cannot unmarshal ICS-27 interchain accounts metadata")
}
```

If that's the case, wouldn't this check here just redundant?

Oh, wait... Or shouldn't you be using here `channel.Version` instead?

Reply:
nice catch, it should be channel.Version :+1:

Reply:
fixed